### DISCUSSION TITLE
[remarkMermaid] crashes hard when parsing invalid syntax

### DISCUSSION THREAD

**Discussion by natarius** [remarkjs/remark#1153] [Q&A] [answered]
I am using react-markdown with remarkMermaid and when users input invalid syntax it causes the whole app to crash.

Even a react error boundary won't catch it :(

is there anything I am missing here? I looked into the source code of react-markdown and couldn't even find any plugin error handling...seems pretty dangerous

**Comment by wooorm** [comment]
- An error boundary should catch plugin errors. You are supposed to handle your errors. That's not something react-markdown should do
- you will have more luck asking on remark-mermaid issue tracker about errors it throws, as it is outside of the remarkjs organization

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @natarius! 👋 
Sorry you ran into a spot of trouble.

It sounds like you are using https://github.com/temando/remark-mermaid?
It hasn't been maintained in a while, the hard error sounds like a bug.

I'd recommend trying https://github.com/remcohaszing/remark-mermaidjs
And set the `errorFallback` option from `remark-mermaidjs` to set the behavior for when mermaid fails.

**Reply by natarius** [reply]
Thanks, that did the trick!
